### PR TITLE
Permit multiple root data directories

### PR DESCRIPTION
**Pull request by CBroz1**
- establishes multiple root abilities within the workflow 
- passes ingestion pytests
- does not pass processing pytests, with errors related to caiman - likely a byproduct of my local install

I've left some cues for caiman installation, but future work would need to clarify and reshape into proper documentation.

[pytest-log.txt](https://github.com/datajoint/workflow-calcium-imaging/files/9748718/pytest-log.txt)

### CODE REVIEW

**Comment by CBroz1** [comment]
Hi @tdincer - I added a new commit to address test fails for `test_find_valid_full_path` and `test_find_root_directory`. I also fetched from upstream, resolving conflicts with the new PrarieView feature on `paths.py`

File: requirements_caiman.txt [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+pyqtgraph
+scikit-image
+scikit-learn
+scipy
+# tensorflow
+tifffile
+tk
+tqdm
+yapf
+caiman @ git+https://github.com/datajoint-company/caiman
```
Comment (kabilar) [inline requirements_caiman.txt:30]: @CBroz1 Is this the strategy that we will be employing to install CaImAn going forward?

Comment (CBroz1) [reply requirements_caiman.txt:30]: Yes, I think this is a good middle-ground to permit pip installation even when supported packages haven't done the same. We may be behind, however, in documenting this method for new users

**Comment by CBroz1** [comment]
Marking as draft pending 
- [ ] Resolved conflicts
- [ ] fully passing pytests

File: requirements_caiman.txt (hunk 1 above)
Comment (tdincer) [reply requirements_caiman.txt:30]: I'm back to caiman installation. I'll resolve this installation issue ASAP.

**Comment by CBroz1** [comment]
Hi @tdincer - Were there other updates here you were looking for before merging? Or do you want to pull these changes into your existing PR and I'll close here?

File: workflow_calcium_imaging/paths.py [hunk 2]
```diff
@@ -22,16 +37,7 @@ def get_scan_image_files(scan_key):
...
-
-    from .pipeline import session
-
-    sess_dir = data_dir / (session.SessionDirectory & scan_key).fetch1("session_dir")
-
-    if not sess_dir.exists():
-        raise FileNotFoundError(f"Session directory not found ({sess_dir})")
-
-    tiff_filepaths = [fp.as_posix() for fp in sess_dir.glob("*.tif")]
+    sess_dir, tiff_filepaths = _find_files_by_type(scan_key, "*.tif")
```
Comment (tdincer) [inline workflow_calcium_imaging/paths.py:40]: A tif file is not always a ScanImage recording but a ScanImage recording is a tif file.

Comment (CBroz1) [reply workflow_calcium_imaging/paths.py:40]: @tdincer Is there a fix you can recommend? I believe my changes here are functionally equivalent to the previous code
```python
tiff_filepaths = [fp.as_posix() for fp in sess_dir.glob("*.tif")]
```
My changes extract this pattern to a common helper function

Comment (tdincer) [reply workflow_calcium_imaging/paths.py:40]: Your changes are fine. I'm just writing one idea but we don't have to apply it now. If the file is a ScanImage file and needs to be read by scanreader, the scanner information can be verified from its header. But reading the header introduces an overhead.

File: tests/__init__.py [hunk 3]
```diff
@@ -33,25 +58,57 @@ def dj_config():
     if pathlib.Path("./dj_local_conf.json").exists():
         dj.config.load("./dj_local_conf.json")
     dj.config["safemode"] = False
+
+    environ_root = os.environ.get("IMAGING_ROOT_DATA_DIR")
+    if environ_root or (environ_root and not isinstance(environ_root, list)):
```
Comment (tdincer) [inline tests/__init__.py:63]: Would this work?

```suggestion
    if (environ_root and not isinstance(environ_root, list)):
```

File: requirements_caiman.txt (hunk 1 above)
Comment (tdincer) [reply requirements_caiman.txt:30]: Let's remove this file since it's not relevant to the PR.

**Review by tdincer — CHANGES_REQUESTED** [review]
Thank you @CBroz1! It looks good in overall. I'd suggest not to include the caiman requirements file since it's not properly tested and not quite relevant to this PR.

On the other hand, I'll submit an issue on the session directory definition.

File: requirements_caiman.txt (hunk 1 above)
Comment (CBroz1) [reply requirements_caiman.txt]: Sure - I've removed this requirements file

**Comment by CBroz1** [comment]
Hi @tdincer - in my last few commits, I've also added the pre-commit file that has been implemented across other Elements. I've separated all formatting/linting changes into individual commits. If anything looks like it needs edits to you, I can revert my `Markdown linting` and `Run pre-commit` commits to prioritize merging sooner rather than later

**Review by tdincer — APPROVED** [review]

**Pull request merged**